pkg/slayers/path/scion/base.go

```diff
@@ -117,6 +137,7 @@ func (s *Base) DecodeFromBytes(data []byte) (r error) {
 // @ ensures  e == nil ==> s.Mem()
 // @ ensures  e == nil ==> s.Len() == old(s.Len())
 // @ ensures  e == nil ==> s.getNumINF() == old(s.getNumINF())
+// @ ensures  e == nil ==> unfolding s.Mem() in 0 <= s.PathMeta.CurrINF && s.PathMeta.CurrINF < 3
```

Comment:
pure function?

Reply:
It is not used anywhere else by itself and it isn't that long, or is it?